### ISSUE TITLE
databases created in 1.2.2 cannot be backed up and restored in 1.5.2

### ISSUE THREAD

**Issue by ivanpricewaycom** [influxdata/influxdb#9780]
Hi there

From influx 1.2.2 we do a complete backup of all our databases, first (all) the meta, then all the data.
On a fresh instance of 1.5.2 we do an 'offline' restore of all the meta/databases: everything appears to work fine, the data has been transferred and is available via queries etc.

On the 1.5.2 instance we now perform a "-portable" backup of a single database, and then drop the database. The files appear to be correctly written.
We then perform a restore of those files (again using -portable). The restore fails with:

```
$ influxd restore -portable -database sla_cvlili /root/sla_cvlili/
2018/04/26 15:38:15 error updating meta: updating metadata on influxd service failed: err=read tcp 127.0.0.1:39456->127.0.0.1:8088: read: connection reset by peer, n=16
restore: updating metadata on influxd service failed: err=read tcp 127.0.0.1:39456->127.0.0.1:8088: read: connection reset by peer, n=16
$
```

in the influx server logs we see one single line:
```
2018-04-26T15:39:58.932722Z	info	failed to decode meta: proto: meta.ShardGroupInfo: illegal tag 0 (wire type 0)	{"log_id": "07heI8BW000", "service": "snapshot"}
```

we are also now incapable of doing a -portable backup for new databases created in the 1.5.2 instance. the act of restoring the original 1.2.2 data appears to have broken the 1.52 instance.

If we start from scratch without performing the original restore of the 1.2.2 data the -portable backup/restore works as expected.

is there a way we can better understand the error:
failed to decode meta: proto: meta.ShardGroupInfo: illegal tag 0 (wire type 0) 
?

thanks for any ideas

appendicies:
```
# backup procedure on 1.2.2:
influxd backup -database $database -retention forever -since $INCR_SINCE_DATE $INFLUX_BACKUP_DIR

# restore procedure on 1.5.2 (offline):
influxd restore -metadir $INFLUX_DATA_DIR/meta/ $INFLUX_BACKUP_DIR/metastore/
for database in $(cat $INFLUX_BACKUP_DIR/databases); do
    influxd restore -database $database -datadir  $INFLUX_DATA_DIR/data/ $INFLUX_BACKUP_DIR
done

```

**Comment by ivanpricewaycom** [comment]
Update:
to help reproduce this I attach a backup of a single-row database ("foo") from our 1.2.2 instance.

This database was created with the following commands:

```
create database foo
use foo
INSERT treasures,captain_id=pirate_king value=2
exit
```
backup was made using:
```
influxd backup -database foo foo
```
attempt to restore on influx 1.5.2:

```
$ influxd restore -online -db foo -newdb foo foo
2018/04/27 09:53:25 Using metastore snapshot: foo/meta.00
2018/04/27 09:53:25 error updating meta: updating metadata on influxd service failed: err=read tcp 127.0.0.1:39236->127.0.0.1:8088: read: connection reset by peer, n=16
restore: updating metadata on influxd service failed: err=read tcp 127.0.0.1:39236->127.0.0.1:8088: read: connection reset by peer, n=16
$
```

The log appears in influx:
```
failed to decode meta: proto: meta.ShardGroupInfo: illegal tag 0 (wire type 0)	{"log_id": "07ibSndG000", "service": "snapshot"}
```

This can be replayed easily using docker:
```
cd && unzip foo.zip

docker run -it -d -p 8086:8086  -e INFLUX_BACKUP_DIR=/var/lib/influxdb/backup -e INFLUX_DATA_DIR=/var/lib/influxdb -v `pwd`/foo:/var/lib/influxdb/backup --name=influxdb influxdb:1.5.2

docker exec -it influxdb bash
# once in docker execute:
influxd restore -online $INFLUX_BACKUP_DIR
```


Foo database backup is attached. 
It appears that Influx 1.5.2 online mode restore is not compatible with backups made by Influx 1.2. I haven't tested backups made by 1.3/1.4. non-portable backups created in and exported from 1.5.2 can be restored using -online mode to the same 1.5.2 instance.

Any ideas on how to get more info regarding:
failed to decode meta: proto: meta.ShardGroupInfo: illegal tag 0 (wire type 0)	{"log_id": "07ibSndG000", "service": "snapshot"}
would be appreciated.


[foo.zip](https://github.com/influxdata/influxdb/files/1954637/foo.zip)

**Comment by HeathKang** [comment]
Hi, I used influxdb 1.2 to backup and influxdb 1.5.2 to restore.
But my restore order is ```influxd restore -metadir /influxdb/meta/  /backup/meta``` ```influxd restore -database mydatabase /influxdb/data/ backup/data/```. Then I change influxdb conf ```[data] index='tsi1'```and use influx_inspect to convert data structure from TSM to TSI ```influx_inspect -database mydatabase -datadir /influxdb/data/ -waldir /influxdb/wal/```. It worked well. I think maybe restore portable was not worked because influxdb 1.2 backup didn't have portable option.

**Comment by ivanpricewaycom** [comment]
Hi Heath, thanks for your response.

That's interesting regarding the conversion of indexes, however this would also be an offline backup/restore, in the sense that you can't be collecting data during the process. It's very useful though thanks for that.

regarding the restore portable, it was influx 1.5.2 that did the portable backup so it should be able to restore a backup it made itself.  For me there is a bug with portable backups in 1.5.2 (as evidenced with the docker reproduce above), I wouldn't trust online backups in production in the current state for the open source version at least.

to summarise:
1.2.2  backup -> 1.5.2 (offline restore ('works')) -> 1.5.2 portable backup -> (new 1.5.2) portable restore fails
and
1.2.2  backup -> 1.5.2 (online restore) fails  (see docker example)

**Comment by teu** [comment]
Having the same problem with 1.4 and 1.5.3. I upgraded influx to 1.5.3 prior to the `-portable` backup but this still didn't help.

```
ts=2018-06-11T09:56:46.298458Z lvl=info msg="failed to decode meta: proto: meta.DatabaseInfo: illegal tag 0 (wire type 0)" log_id=08c_5UU0000 service=snapshot
```

**Comment by galindro** [comment]
Same problem here. I'm making backups from 1.5.2 OSS to restore to 1.6.4 OSS.

Backup command:
```bash
influxd backup -host myhost_to_backup:8088 -portable /influxdb_prod_backup/
```
Restore command:
```bash
influxd restore -host myhost_to_restore -portable /influxdb_prod_backup/
```

Error:
```bash
ts=2018-11-02T15:18:58.265897Z lvl=info msg="failed to decode meta: proto: meta.ShardGroupInfo: illegal tag 0 (wire type 0)" log_id=0BXFl9yW000 service=snapshot
```

**Comment by galindro** [comment]
I've tested making backup from 1.6.4 to 1.6.4, using the same aforementioned commands for backup and restore and the problem persists

**Comment by galindro** [comment]
Guys, it seems to be something related to docker networking, not with influxdb itself. I've made a test running an influxdb 1.6.4 container sharing the volume that contains backup files made from influxdb 1.6.4. Withing it, I've executed the `influxd restore` command and it worked like a charm.

**Comment by ErnoL** [comment]
I had similar case but the error I got when running restore to remote influxdb docker instance was:
```
$ influxd restore -db foo -portable -host 192.168.12.13:8088 -newdb bar ~/backup/
error updating meta: DB metadata not changed. database may already exist
restore: DB metadata not changed. database may already exist
```
The `-newdb bar` did not exist on the remote influxdb instance and the logs only had:
`lvl=info msg="failed to decode meta: proto: meta.ShardGroupInfo: illegal tag 0 (wire type 0)"`
Found @galindro 's helpful comment. Moving the backup files to the remote and running locally worked like it should:
```
$ sudo docker exec -it container_id /bin/bash
# mkdir /tmp/backup; exit
$ sudo docker cp ~/tmp/portable_backup/. container_id:/tmp/backup
$ sudo docker exec -it container_id /bin/bash
# influxd restore -db foo -portable -newdb bar /tmp/backup/
Restoring shard 367 live from backup ***.tar.gz
...
```
